storage/remote/queue_manager.go

```diff
@@ -632,6 +659,34 @@ func (t *QueueManager) Stop() {
 	t.metrics.unregister()
 }
 
+// updateCheckpoint gets the current segment number read from the watcher and stores it to a file.
+func (t *QueueManager) updateCheckpoint() {
+	// Skip updating the file if the segment hasn't changed.
+	currentSegement := t.watcher.CurrentSegment()
+	if currentSegement == t.prevSegment {
+		return
+	}
+
+	segment := strconv.Itoa(currentSegement)
+	checkpointDir := filepath.Join(t.tsdbDir, "remote", t.client().Name())
+	err := os.MkdirAll(checkpointDir, 0777)
+	if err != nil {
+		level.Warn(t.logger).Log("msg", "error creating temporary remote write checkpoint directory", "error", err)
```

Comment:
Should this return on error? Ditto with WriteFile/Replace failing.

Reply:
yep